src/components/DragonDeploy/DragonDeploy.tsx

```diff
@@ -0,0 +1,188 @@
+
+import CloseCrossIcon from '@/assets/icons/close-cross.svg'
+import { Button } from '@/components/common/buttons'
+import CostEstimatesToolTip from '@/components/CostEstimatesToolTip'
+import { type Commit, type File, getDeploymentBranchFiles, hasIndexFile, uploadFiles } from '@/lib/dragondeploy'
+import { useGlobalStore } from '@/stores/globalStore'
+import { Deployment } from '@/types/repository'
+
+export default function DragonDeploy() {
+  const [isOpen, setIsOpen] = useState(false)
+  const [isDeploying, setIsDeploying] = useState(false)
+  const [isProcessing, setIsProcessing] = useState(true)
+  const [files, setFiles] = useState<File[]>([])
+  const [commit, setCommit] = useState<Commit>()
+  const [deployedTxId, setDeployedTxId] = useState('')
+  const [currentDeployment, setCurrentDeployment] = useState<Deployment>()
+  const [uploadPercent, setUploadPercent] = useState(0)
+  const [currentBranch, selectedRepo, branchState, branchActions, addDeployment] = useGlobalStore((state) => [
+    state.branchState.currentBranch,
+    state.repoCoreState.selectedRepo.repo,
+    state.branchState,
+    state.branchActions,
+    state.repoCoreActions.addDeployment
+  ])
+
+  function closeModal() {
+    setIsOpen(false)
+  }
+
+  async function deploy() {
+    if (selectedRepo && commit && files.length > 0) {
+      setIsDeploying(true)
+      if (!hasIndexFile(files)) {
+        toast.error("No 'index.html' file found.")
+        setIsDeploying(false)
+        return
+      }
+
+      setUploadPercent(0)
+      try {
+        const response = await uploadFiles(files, commit, selectedRepo, setUploadPercent)
+        const deployment = await addDeployment({
+          txId: response.id,
+          commitMessage: commit.message,
+          commitOid: commit.oid
+        })
+        setCurrentDeployment(deployment)
+        setDeployedTxId(response.id)
+        toast.success('Deployed successfully')
+      } catch (err) {
+        toast.error('Deploy failed')
+      }
+      setIsDeploying(false)
+    }
+  }
+
+  useEffect(() => {
+    if (branchState.branchList.length === 0) {
+      branchActions.listBranches()
+    }
+  }, [branchState])
+
+  useEffect(() => {
+    if (selectedRepo && isOpen) {
+      ;(async () => {
+        setIsProcessing(true)
+        try {
+          const { files: branchFiles, commit: latestCommit } = await getDeploymentBranchFiles(
+            selectedRepo,
+            currentBranch
+          )
+          const deployment = selectedRepo.deployments.find((deployment) => deployment.commitOid === latestCommit.oid)
+          setFiles(branchFiles)
+          setCommit(latestCommit)
+          setCurrentDeployment(deployment)
+          setDeployedTxId(deployment?.txId ?? '')
+          setIsProcessing(false)
+        } catch (error) {
+          toast.error((error as any).message)
+        }
+      })()
+    }
+  }, [selectedRepo?.deploymentBranch, isOpen])
+
+  if (selectedRepo?.deploymentBranch === '') return <></>
```

Comment:
lets return null here. makes much sense to indicate component is returning nothing.